util/data/msgreply.c

```diff
@@ -988,6 +988,35 @@ int edns_opt_append(struct edns_data* edns, struct regional* region,
 	return 1;
 }
 
+int edns_opt_append_ede(struct edns_data* edns, struct regional* region,
+	sldns_ede_code code, const char *txt)
+{
+	struct edns_option** prevp;
+	struct edns_option* opt;
+	size_t txt_len = txt ? strlen(txt) : 0;
+
+	/* allocate new element */
+	opt = (struct edns_option*)regional_alloc(region, sizeof(*opt));
+	if(!opt)
+		return 0;
+	opt->next = NULL;
+	opt->opt_code = LDNS_EDNS_EDE;
+	opt->opt_len = txt_len + sizeof(uint16_t);
+	opt->opt_data = regional_alloc(region, txt_len + sizeof(uint16_t));
+	if(!opt->opt_data)
+		return 0;
+	sldns_write_uint16(opt->opt_data, (uint16_t)code);
+	if (txt_len)
+		strncpy(opt->opt_data + 2, txt, txt_len);
```

Comment:
```suggestion
		memmove(opt->opt_data + 2, txt, txt_len);
```
You want to always copy that length of data, so memmove. Not use the broken strncpy behaviour, or have stripped off end with zero terminator, or shorter string with uninitialised memory.

Reply:
resolved in 84da240